.github/workflows/ci.yml

```diff
@@ -0,0 +1,100 @@
+name: ci
+
+on:
+  push:
+    branches:
+      - master
+      - "dependabot/**"
+      - "snyk-**"
+  pull_request:
+    branches: [master]
+    types: [opened, reopened]
+
+jobs:
+  install:
+    # NOTE: Netlify uses ubuntu 16.08 but Github Actions does not offer it by default.
+    # Hence, we default to the latest version.
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Use Node.js
+        uses: actions/setup-node@v1
+        with:
+          node-version: "14.x"
+      - name: Cache Node.js modules
+        uses: actions/cache@v2
+        with:
+          # npm cache files are stored in `~/.npm` on Linux/macOS
+          path: ~/.npm
+          key: ${{ runner.OS }}-node-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.OS }}-node-
+            ${{ runner.OS }}-
+      - run: npm ci
+
+  lint:
+    needs: install
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Use Node.js
+        uses: actions/setup-node@v1
+        with:
+          node-version: "14.x"
+      - name: Load Node.js modules
+        uses: actions/cache@v2
+        with:
+          # npm cache files are stored in `~/.npm` on Linux/macOS
+          path: ~/.npm
+          key: ${{ runner.OS }}-node-${{ hashFiles('**/package-lock.json') }}
+      - run: npm ci
+      - run: npm run lint-fix
+      - run: npm run format-fix
+
+  build:
+    needs: lint
+    runs-on: ubuntu-latest
+    env:
+      CI: false
+    steps:
+      - uses: actions/checkout@v2
+      - name: Use Node.js
+        uses: actions/setup-node@v1
+        with:
+          node-version: "14.x"
+      - name: Load Node.js modules
+        uses: actions/cache@v2
+        with:
+          # npm cache files are stored in `~/.npm` on Linux/macOS
+          path: ~/.npm
+          key: ${{ runner.OS }}-node-${{ hashFiles('**/package-lock.json') }}
+      - run: npm ci
+      - run: npm run build
+
+  test:
+    needs: build
+    runs-on: ubuntu-latest
+    environment: staging
+    env:
+      CYPRESS_RECORD_KEY: ${{secrets.CYPRESS_RECORD_KEY}}
+      CYPRESS_BASEURL: ${{secrets.CYPRESS_BASEURL}}
+      CYPRESS_COOKIE_NAME: ${{secrets.CYPRESS_COOKIE_NAME}}
+      CYPRESS_COOKIE_VALUE: ${{secrets.CYPRESS_COOKIE_VALUE}}
+      CYPRESS_TEST_REPO_NAME: ${{secrets.CYPRESS_TEST_REPO_NAME}}
+      USERNAME: ${{secrets.USERNAME}}
+      PERSONAL_ACCESS_TOKEN: ${{secrets.PERSONAL_ACCESS_TOKEN}}
+      E2E_COMMIT_HASH: ${{secrets.E2E_COMMIT_HASH}}
+    steps:
+      - uses: actions/checkout@v2
+      - name: Use Node.js
+        uses: actions/setup-node@v1
+        with:
+          node-version: "14.x"
+      - name: Load Node.js modules
+        uses: actions/cache@v2
+        with:
+          # npm cache files are stored in `~/.npm` on Linux/macOS
+          path: ~/.npm
+          key: ${{ runner.OS }}-node-${{ hashFiles('**/package-lock.json') }}
+      - run: npm ci
+      - run: npm run test:ci
```

Comment:
also, something i'm not entirely sure about is _how_ this works - cypress works by obtaining a website url and then invoking events on the dom. this seems like it should fail (and it does on local) as hte website is not running so i'm not sure how it worked previously. 

if anyone can explain this, it'd be greatly appreciated!

Reply:
This is a good catch - I don't think GitHub Actions runners have a browser by default. We might want to try using the [cypress-io/github-action@v2](https://docs.cypress.io/guides/continuous-integration/github-actions#Cypress-GitHub-Action) plugin to run the tests.

Reply:
i actually saw this before and i was under the impression that it did the same thing as what we're doing here, which is why i omitted it. 

in fact, the runner [does](https://github.com/actions/virtual-environments/blob/main/images/linux/Ubuntu2004-Readme.md) have chrome!

Reply:
You're right - it does. So actually we can proceed with the current CI configs (without cypress-io/github-action@v2)

Reply:
Also for summary, @seaerchin and I had an offline discussion. We clarified that the e2e tests work on the live staging CMS, and so there is no need for the CI runner to run a local version of the CMS.